### PR TITLE
Pinned versions and CI setup

### PR DESCRIPTION
**Pull request by mikek**
Pinned dependency versions and CI setup refactor.

 * Fix imports for django-polymorphic>=0.8
 * Pin requirements in setup.py
 * Simplify tox setup	
 * Refactor Travis setup to use tox environments list
 * Update README with some version compatibility hints

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,61 +1,67 @@
...
-    - python: 2.7
-      env: DJANGO="Django>=1.8,<1.9" EASY_THUMBNAILS="easy-thumbnails>2.0" CUSTOM_IMAGE="filer.test_utils.custom_image.models.Image"
-    - python: 3.3
-      env: DJANGO="Django>=1.8,<1.9" EASY_THUMBNAILS="easy-thumbnails>2.0" CUSTOM_IMAGE="filer.test_utils.custom_image.models.Image"
-    - python: 3.4
-      env: DJANGO="Django>=1.8,<1.9" EASY_THUMBNAILS="easy-thumbnails>2.0" CUSTOM_IMAGE="filer.test_utils.custom_image.models.Image"
+    - env: TOX_ENV=flake8
+    - env: TOX_ENV=py34-django_master-thumbs2x
+    - env: TOX_ENV=py33-django_master-thumbs2x
+    - env: TOX_ENV=py27-django_master-thumbs2x
```
Comment (mkoistinen) [inline .travis.yml]: Why are these builds allowed as failures?

File: README.rst [hunk 2]
```diff
@@ -22,16 +22,21 @@ Dependencies
...
-**Django >= 1.6** is supported together with `django-polymorphic`_ >= 0.5.4
+Please note, there are some compatibility constraints that we can not enforce
+through the `setup.py`. Here are the most important of them::
 
-**Django >= 1.7** is supported together with `django-polymorphic`_ >= 0.5.6
+    Django | django-polymorphic | django-mptt
+    ------ | ------------------ | -----------
+    1.6    | >=0.5.4,           | >=0.6,<0.8
+    1.7    | >=0.5.6            | >=0.6,<0.8
+    1.8    | >=0.7              | >=0.7
```
Comment (mkoistinen) [inline README.rst]: Nice!

**Comment by mkoistinen** [comment]
Just a couple of comments, but otherwise, LGTM.

File: .travis.yml (hunk 1 above)
Comment (mikek) [inline .travis.yml]: Django master is based on Django 1.9 that is not supported yet.
@mkoistinen, do you want flake8 to be a strict requirement for a successful build also?

Comment (mkoistinen) [inline .travis.yml]: flake8 is strictly required and is in-fact the first build we do everywhere else. However, in this project, I've configured flake8 to not include a line-length test, we will have to change this at some point though.

File: .travis.yml [hunk 3]
```diff
@@ -1,61 +1,67 @@
...
-      env: DJANGO="Django>=1.8,<1.9" EASY_THUMBNAILS="easy-thumbnails>2.0"
-    - python: 3.4
-      env: DJANGO="Django>=1.8,<1.9" EASY_THUMBNAILS="easy-thumbnails>2.0"
-    - python: 2.7
-      env: DJANGO="Django>=1.8,<1.9" EASY_THUMBNAILS="easy-thumbnails>2.0" CUSTOM_IMAGE="filer.test_utils.custom_image.models.Image"
-    - python: 3.3
-      env: DJANGO="Django>=1.8,<1.9" EASY_THUMBNAILS="easy-thumbnails>2.0" CUSTOM_IMAGE="filer.test_utils.custom_image.models.Image"
-    - python: 3.4
-      env: DJANGO="Django>=1.8,<1.9" EASY_THUMBNAILS="easy-thumbnails>2.0" CUSTOM_IMAGE="filer.test_utils.custom_image.models.Image"
+    - env: TOX_ENV=flake8
```
Comment (mkoistinen) [inline .travis.yml]: Should be required and the first test to run.

**Comment by yakky** [comment]
Why pinning polymorphic and not mptt (#641)?

**Comment by mikek** [comment]
@yakky I've read the discussion and see your point. However, the real limit on mptt version depends on Django version used in a project. So, we do not put hard limits in setup.py (but we do in tests) and let used pin/limit mptt version in his project. I Think this is a price we have to pay.

**Comment by cyberstar** [comment]
LGTM

**Comment by mkoistinen** [comment]
@mikek I think you're doing the mptt pinning correctly. LGTM.

**Pull request merged**